frontend/src/components/mine/Permit/MinePermitInfo.js

```diff
@@ -107,48 +99,55 @@ const getPermittees = (partyRelationships, permits) =>
 const getPermitteeName = (permittees) =>
   permittees[0] ? permittees[0].party.party_name : Strings.EMPTY_FIELD;
 
-const transformRowData = (permits, partyRelationships) => {
-  const latest = permits[0];
-  const first = permits[permits.length - 1];
+const transformRowData = (permit, partyRelationships) => {
+  const latestAmendment = permit.amendments[0];
+  const firstAmendment = permit.amendments[permit.amendments.length - 1];
 
-  const permittees = getPermittees(partyRelationships, permits);
+  const permittees = getPermittees(partyRelationships, permit);
   const permitteeName =
     partyRelationships.length === 0 ? Strings.LOADING : getPermitteeName(permittees);
 
   return {
-    key: latest.permit_guid,
-    lastAmended: formatDate(latest.issue_date),
-    permitNo: latest.permit_no || Strings.EMPTY_FIELD,
-    firstIssued: formatDate(first.issue_date) || Strings.EMPTY_FIELD,
+    key: permit.permit_guid,
+    lastAmended:
+      (latestAmendment && latestAmendment.issue_date && formatDate(latestAmendment.issue_date)) ||
+      Strings.EMPTY_FIELD,
+    permitNo: permit.permit_no || Strings.EMPTY_FIELD,
+    firstIssued:
+      (firstAmendment && firstAmendment.issue_date && formatDate(firstAmendment.issue_date)) ||
+      Strings.EMPTY_FIELD,
     permittee: permitteeName,
-    authorizationEndDate: latest.authorization_end_date
-      ? formatDate(latest.authorization_end_date)
-      : Strings.EMPTY_FIELD,
-    permittees,
-    status: Strings.EMPTY_FIELD,
+    authorizationEndDate:
+      latestAmendment && latestAmendment.authorization_end_date
+        ? formatDate(latestAmendment.authorization_end_date)
+        : Strings.EMPTY_FIELD,
+    amendments: permit.amendments,
+    status: permit.permit_status_code,
   };
 };
 
-const transformChildRowData = (permittee, record) => ({
-  key: record.key,
-  permitNo: record.permitNo,
-  Date: permittee.start_date,
-  permittee: permittee.party.name,
+const transformChildRowData = (amendment, record, amendmentNumber) => ({
+  amendmentNumber,
+  receivedDate:
+    (amendment.received_date && formatDate(amendment.received_date)) || Strings.EMPTY_FIELD,
+  issueDate: (amendment.issue_date && formatDate(amendment.issue_date)) || Strings.EMPTY_FIELD,
+  authorizationEndDate:
+    (amendment.authorization_end_date && formatDate(amendment.authorization_end_date)) ||
+    Strings.EMPTY_FIELD,
   description: Strings.EMPTY_FIELD,
 });
 
 export const MinePermitInfo = (props) => {
-  const groupedPermits = Object.values(groupPermits(props.mine.mine_permit));
   const amendmentHistory = (record) => {
-    const childRowData = record.permittees.map((permittee) =>
-      transformChildRowData(permittee, record)
+    const childRowData = record.amendments.map((amendment, index) =>
+      transformChildRowData(amendment, record, record.amendments.length - index)
```

Comment:
So, it looks like we can change `transformChildRowData` slightly and remove a lot of the busy code here. Plus, it looks like record is an unused argument in `transformChildRowData`?
```javascript
const transformChildRowData = (amendment, index, amendments) => ({
  amendmentNumber: amendments.length - index,
 // the rest
})

const amendmentHistory = (record) => {
  const childRowData = record.amendments.map(transformChildRowData));
  // the rest
};
```

Reply:
Cool, tweaked!